runtime/doc/treesitter.txt

```diff
@@ -584,57 +590,63 @@ add_directive({name}, {handler}, {force})                    *add_directive()*
     `metadata[capture_id].key = value`
 
     Parameters: ~
-        {name}     the name of the directive, without leading #
-        {handler}  the handler function to be used signature will be (match,
-                   pattern, bufnr, predicate, metadata)
+        {name}     (string) Name of the directive, without leading #
+        {handler}  (function) with signature (match, pattern, bufnr,
+                   predicate, metadata)
 
 add_predicate({name}, {handler}, {force})                    *add_predicate()*
     Adds a new predicate to be used in queries
 
     Parameters: ~
-        {name}     the name of the predicate, without leading #
-        {handler}  the handler function to be used signature will be (match,
-                   pattern, bufnr, predicate)
+        {name}     (string) Name of the predicate, without leading #
+        {handler}  (function) with Signature (match, pattern, bufnr,
```

Comment:
Does it make sense to write this as `fun(match:string, pattern:string, bufnr:number, predicate:function)` as that seems to be supported by Emmylua?

Reply:
If this is supported by our gendoc script, yes.